<?xml version="1.0" ?>
<sdf version="1.8">
  <model name="f330">
    <link name="base_link">
      <enable_wind>true</enable_wind>
      <pose>0 0 0 0 -0 0</pose>
      <inertial>
      <pose>0 0 0 0 -0 0</pose>
      <mass>1.5</mass>
      <inertia>
          <ixx>0.025</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.009</iyy>
          <iyz>0</iyz>
          <izz>0.033</izz>
      </inertia>
      </inertial>
      <visual name="f330_visual">
        <geometry>
          <mesh>
            <uri>model://f330/meshes/model.dae</uri>
            <submesh>
              <name>f330</name>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0 1.0</diffuse>
          <specular>0.0 0.0 0.0 1.0</specular>
          <pbr>
            <metal/>
          </pbr>
        </material>
      </visual>
      <collision name="f330_collision">
        <geometry>
          <mesh>
            <uri>model://f330/meshes/model.dae</uri>
          </mesh>
        </geometry>
        <surface>
          <contact>
            <collide_bitmask>0x01</collide_bitmask>
          </contact>
        </surface>
      </collision>
    </link>

    <!-- IMU -->
    <model name='imu'>
        <link name="internal">
            <inertial>
                <mass>0.005</mass>
                <inertia>
                    <ixx>8.33e-06</ixx>
                    <ixy>0</ixy>
                    <ixz>0</ixz>
                    <iyy>8.33e-06</iyy>
                    <iyz>0</iyz>
                    <izz>8.33e-06</izz>
                </inertia>
            </inertial>
            <sensor name='imu' type='imu'>
                <always_on>1</always_on>
                <update_rate>250</update_rate>
                <imu>
                    <angular_velocity>
                        <x>
                            <noise type='gaussian'>
                                <mean>0</mean>
                                <stddev>0.009</stddev>
                                <bias_mean>0.00075</bias_mean>
                                <bias_stddev>0.005</bias_stddev>
                                <dynamic_bias_stddev>0.00002</dynamic_bias_stddev>
                                <dynamic_bias_correlation_time>400.0</dynamic_bias_correlation_time>
                                <precision>0.00025</precision>
                            </noise>
                        </x>
                        <y>
                            <noise type='gaussian'>
                                <mean>0</mean>
                                <stddev>0.009</stddev>
                                <bias_mean>0.00075</bias_mean>
                                <bias_stddev>0.005</bias_stddev>
                                <dynamic_bias_stddev>0.00002</dynamic_bias_stddev>
                                <dynamic_bias_correlation_time>400.0</dynamic_bias_correlation_time>
                                <precision>0.00025</precision>
                            </noise>
                        </y>
                        <z>
                            <noise type='gaussian'>
                                <mean>0</mean>
                                <stddev>0.009</stddev>
                                <bias_mean>0.00075</bias_mean>
                                <bias_stddev>0.005</bias_stddev>
                                <dynamic_bias_stddev>0.00002</dynamic_bias_stddev>
                                <dynamic_bias_correlation_time>400.0</dynamic_bias_correlation_time>
                                <precision>0.00025</precision>
                            </noise>
                        </z>
                    </angular_velocity>
                    <linear_acceleration>
                        <x>
                            <noise type='gaussian'>
                                <mean>0</mean>
                                <stddev>0.021</stddev>
                                <bias_mean>0.05</bias_mean>
                                <bias_stddev>0.0075</bias_stddev>
                                <dynamic_bias_stddev>0.000375</dynamic_bias_stddev>
                                <dynamic_bias_correlation_time>175.0</dynamic_bias_correlation_time>
                                <precision>0.005</precision>
                            </noise>
                        </x>
                        <y>
                            <noise type='gaussian'>
                                <mean>0</mean>
                                <stddev>0.021</stddev>
                                <bias_mean>0.05</bias_mean>
                                <bias_stddev>0.0075</bias_stddev>
                                <dynamic_bias_stddev>0.000375</dynamic_bias_stddev>
                                <dynamic_bias_correlation_time>175.0</dynamic_bias_correlation_time>
                                <precision>0.005</precision>
                            </noise>
                        </y>
                        <z>
                            <noise type='gaussian'>
                                <mean>0</mean>
                                <stddev>0.021</stddev>
                                <bias_mean>0.05</bias_mean>
                                <bias_stddev>0.0075</bias_stddev>
                                <dynamic_bias_stddev>0.000375</dynamic_bias_stddev>
                                <dynamic_bias_correlation_time>175.0</dynamic_bias_correlation_time>
                                <precision>0.005</precision>
                            </noise>
                        </z>
                    </linear_acceleration>
                </imu>
            </sensor>
        </link>
    </model>

    <joint name="imu_joint" type="fixed">
        <parent>base_link</parent>
        <child>imu</child>
    </joint>

    <link name='rotor_0'>
        <pose>0.12763552868 -0.12763552868 0.025 0 -0 0</pose>
        <inertial>
        <pose>0 0 0 0 -0 0</pose>
        <mass>0.005</mass>
        <inertia>
            <ixx>9.75e-07</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>4.17041e-05</iyy>
            <iyz>0</iyz>
            <izz>4.26041e-05</izz>
        </inertia>
        </inertial>
        <collision name='rotor_0_collision'>
        <pose>0 0 0 0 -0 0</pose>
        <geometry>
            <cylinder>
            <length>0.005</length>
            <radius>0.1</radius>
            </cylinder>
        </geometry>
        <surface>
            <contact>
            <ode/>
            </contact>
            <friction>
            <ode/>
            </friction>
        </surface>
        </collision>
        <visual name='rotor_0_visual'>
        <pose>0 0 0 0 -0 0</pose>
        <geometry>
            <mesh>
            <scale>0.1 0.1 0.1</scale>
            <uri>model://quadrotor_base/meshes/propeller_ccw.dae</uri>
            </mesh>
        </geometry>
        <material>
            <diffuse>0 0 1 1</diffuse>
            <script>
            <name>Gazebo/Blue</name>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            </script>
        </material>
        <cast_shadows>0</cast_shadows>
        </visual>
        <gravity>1</gravity>
        <velocity_decay/>
    </link>
    <joint name='rotor_0_joint' type='revolute'>
        <child>rotor_0</child>
        <parent>base_link</parent>
        <axis>
        <xyz expressed_in='__model__'>0 0 1</xyz>
        <limit>
            <lower>-1e+16</lower>
            <upper>1e+16</upper>
        </limit>
        <dynamics>
            <spring_reference>0</spring_reference>
            <spring_stiffness>0</spring_stiffness>
        </dynamics>
        </axis>
    </joint>
    <link name='rotor_1'>
        <pose>-0.12763552868 0.12763552868 0.025 0 -0 0</pose>
        <inertial>
        <pose>0 0 0 0 -0 0</pose>
        <mass>0.005</mass>
        <inertia>
            <ixx>9.75e-07</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>4.17041e-05</iyy>
            <iyz>0</iyz>
            <izz>4.26041e-05</izz>
        </inertia>
        </inertial>
        <collision name='rotor_1_collision'>
        <pose>0 0 0 0 -0 0</pose>
        <geometry>
            <cylinder>
            <length>0.005</length>
            <radius>0.1</radius>
            </cylinder>
        </geometry>
        <surface>
            <contact>
            <ode/>
            </contact>
            <friction>
            <ode/>
            </friction>
        </surface>
        </collision>
        <visual name='rotor_1_visual'>
        <pose>0 0 0 0 -0 0</pose>
        <geometry>
            <mesh>
            <scale>0.1 0.1 0.1</scale>
            <uri>model://quadrotor_base/meshes/propeller_ccw.dae</uri>
            </mesh>
        </geometry>
        <material>
            <diffuse>1 0 0 1</diffuse>
            <script>
            <name>Gazebo/Red</name>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            </script>
        </material>
        <cast_shadows>0</cast_shadows>
        </visual>
        <gravity>1</gravity>
        <velocity_decay/>
    </link>
    <joint name='rotor_1_joint' type='revolute'>
        <child>rotor_1</child>
        <parent>base_link</parent>
        <axis>
        <xyz expressed_in='__model__'>0 0 1</xyz>
        <limit>
            <lower>-1e+16</lower>
            <upper>1e+16</upper>
        </limit>
        <dynamics>
            <spring_reference>0</spring_reference>
            <spring_stiffness>0</spring_stiffness>
        </dynamics>
        </axis>
    </joint>
    <link name='rotor_2'>
        <pose>0.12763552868 0.12763552868 0.025 0 -0 0</pose>
        <inertial>
        <pose>0 0 0 0 -0 0</pose>
        <mass>0.005</mass>
        <inertia>
            <ixx>9.75e-07</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>4.17041e-05</iyy>
            <iyz>0</iyz>
            <izz>4.26041e-05</izz>
        </inertia>
        </inertial>
        <collision name='rotor_2_collision'>
        <pose>0 0 0 0 -0 0</pose>
        <geometry>
            <cylinder>
            <length>0.005</length>
            <radius>0.1</radius>
            </cylinder>
        </geometry>
        <surface>
            <contact>
            <ode/>
            </contact>
            <friction>
            <ode/>
            </friction>
        </surface>
        </collision>
        <visual name='rotor_2_visual'>
        <pose>0 0 0 0 -0 0</pose>
        <geometry>
            <mesh>
            <scale>0.1 0.1 0.1</scale>
            <uri>model://quadrotor_base/meshes/propeller_cw.dae</uri>
            </mesh>
        </geometry>
        <material>
            <diffuse>0 0 1 1</diffuse>
            <script>
            <name>Gazebo/Blue</name>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            </script>
        </material>
        <cast_shadows>0</cast_shadows>
        </visual>
        <gravity>1</gravity>
        <velocity_decay/>
    </link>
    <joint name='rotor_2_joint' type='revolute'>
        <child>rotor_2</child>
        <parent>base_link</parent>
        <axis>
        <xyz expressed_in='__model__'>0 0 1</xyz>
        <limit>
            <lower>-1e+16</lower>
            <upper>1e+16</upper>
        </limit>
        <dynamics>
            <spring_reference>0</spring_reference>
            <spring_stiffness>0</spring_stiffness>
        </dynamics>
        </axis>
    </joint>
    <link name='rotor_3'>
        <pose>-0.12763552868 -0.12763552868 0.025 0 -0 0</pose>
        <inertial>
        <pose>0 0 0 0 -0 0</pose>
        <mass>0.005</mass>
        <inertia>
            <ixx>9.75e-07</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>4.17041e-05</iyy>
            <iyz>0</iyz>
            <izz>4.26041e-05</izz>
        </inertia>
        </inertial>
        <collision name='rotor_3_collision'>
        <pose>0 0 0 0 -0 0</pose>
        <geometry>
            <cylinder>
            <length>0.005</length>
            <radius>0.1</radius>
            </cylinder>
        </geometry>
        <surface>
            <contact>
            <ode/>
            </contact>
            <friction>
            <ode/>
            </friction>
        </surface>
        </collision>
        <visual name='rotor_3_visual'>
        <pose>0 0 0 0 -0 0</pose>
        <geometry>
            <mesh>
            <scale>0.1 0.1 0.1</scale>
            <uri>model://quadrotor_base/meshes/propeller_cw.dae</uri>
            </mesh>
        </geometry>
        <material>
            <diffuse>1 0 0 1</diffuse>
            <script>
            <name>Gazebo/Red</name>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            </script>
        </material>
        <cast_shadows>0</cast_shadows>
        </visual>
        <gravity>1</gravity>
        <velocity_decay/>
    </link>
    <joint name='rotor_3_joint' type='revolute'>
        <child>rotor_3</child>
        <parent>base_link</parent>
        <axis>
        <xyz expressed_in='__model__'>0 0 1</xyz>
        <limit>
            <lower>-1e+16</lower>
            <upper>1e+16</upper>
        </limit>
        <dynamics>
            <spring_reference>0</spring_reference>
            <spring_stiffness>0</spring_stiffness>
        </dynamics>
        </axis>
    </joint>
  </model>
</sdf>
